example/shared/src/commonMain/kotlin/viewmodel/featurelist/FeatureListViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.splendo.kaluga.example.shared.viewmodel.featureList
+
+import com.splendo.kaluga.architecture.navigation.NavigationAction
+import com.splendo.kaluga.architecture.navigation.NavigationBundle
+import com.splendo.kaluga.architecture.navigation.Navigator
+import com.splendo.kaluga.architecture.observable.observableOf
+import com.splendo.kaluga.architecture.viewmodel.NavigatingViewModel
+
+sealed class FeatureListNavigationAction : NavigationAction<Nothing>(null) {
+
+    object Location : FeatureListNavigationAction()
+    object Permissions : FeatureListNavigationAction()
+    object Alerts : FeatureListNavigationAction()
+    object LoadingIndicator : FeatureListNavigationAction()
+    object Architecture:  FeatureListNavigationAction()
+}
+
+sealed class Feature(val title: String) {
```

Comment:
thinking out loud: imagine this is an enum, below list would be `observableOf(Feature.values())`. If these enum values provide a `NavigationAction` property 'navigation', we get `navigator.navigate(feature.navigation)`.

Not so much saying this code needs to be rewritten (probably more demonstrative like this) but curious how you'd judge my code if I wrote it like that?

Reply:
Yes that would be fine IMO. Maybe even better than whats there now